### PR TITLE
ITSU-67: Initial commit open-iscsi

### PR DESCRIPTION
**Pull request by SlothOfAnarchy**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# open-iscsi
+
+This role installs and configures open-iscsi, an iSCSI initiator.
+
+## Requirements
+
+Debian
```
Comment (Mr-Pi) [inline README.md]: Please write something like:

Any debian based system should work, tested on debian 9.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+| `open_iscsi_nr_sessions`               | `1`                 | Number of sessions per iface record. It may be useful to use more than one session for multipath setups.                                                                                                                               |
+| `open_iscsi_fast_abort`                | `True`              | Do not respond to PDUs after a management function like an `ABORT`.                                                                                                                                                                    |
+
+### Login authentication
+These login credentials for logging into targets are only used when `open_iscsi_authentication` is `True` and have to be defined in username/password pairs.
+You can choose to define one or both of the pairs.
+| Variable                      | Description                                                |
```
Comment (Mr-Pi) [inline README.md:46]: Please add a newline for proper markdown rendering.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+|:------------------------------|:-----------------------------------------------------------|
+| `open_iscsi_auth_username`    | Username to authenticate at the target                     |
+| `open_iscsi_auth_password`    | Password to authenticate at the target                     |
+| `open_iscsi_auth_username_in` | Username to authenticate the target against this initiator |
+| `open_iscsi_auth_password_in` | Password to authenticate the target against this initiator |
+
+### Discovery authentication
+These login credentials for the discovery phase are only used when `open_iscsi_discovery_authentication` is `True` and have to be defined in username/password pairs.
+You can choose to define one or both of the pairs.
```
Comment (Mr-Pi) [inline README.md:55]: Please add a newline for proper markdown rendering.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+E.g. `CRC32C,None` asks the target to check the digest, but does not enforce it.
+
+## Example Playbook
+
+```yml
+- hosts: hypervisor
+  roles:
+  - open_iscsi_authentication: True
+    open_iscsi_initiator_name: 'iqn.2013-07.de.uni-stuttgart.stuvus.iscsi:hypervisor01'
+	open_iscsi_auth_username: hypervisor01
```
Comment (Mr-Pi) [inline README.md]: Take care of mixed tabs and blanks.

File: templates/iscsid.conf.j2 [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+node.session.iscsi.FirstBurstLength = {{ open_iscsi_first_burst_length }}
+node.session.iscsi.MaxBurstLength = {{ open_iscsi_max_burst_length }}
+node.conn[0].iscsi.MaxRecvDataSegmentLength = {{ open_iscsi_max_recv_length }}
+node.conn[0].iscsi.MaxXmitDataSegmentLength = {{ open_iscsi_max_xmit_length }}
+discovery.sendtargets.iscsi.MaxRecvDataSegmentLength = {{ open_iscsi_discovery_max_recv_length }}
+node.conn[0].iscsi.HeaderDigest = {{ open_iscsi_header_digest }}
+node.conn[0].iscsi.DataDigest = {{ open_iscsi_data_digest }}
+node.session.nr_sessions = {{ open_iscsi_nr_sessions }}
+node.session.iscsi.FastAbort = {% if open_iscsi_fast_abort %}Yes{% else %}No{% endif %}
+
```
Comment (Mr-Pi) [inline templates/iscsid.conf.j2]: Remove newline.

File: templates/iscsid.conf.j2 [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+	{% endif %}
+	{% if open_iscsi_discovery_auth_username_in is defined and open_iscsi_discovery_auth_password_in is defined -%}
+discovery.sendtargets.auth.username_in = {{ open_iscsi_discovery_auth_username_in }}
+discovery.sendtargets.auth.password_in = {{ open_iscsi_discovery_auth_password_in }}
+	{% endif %}
+{% endif %}
+
+# Timeouts
+node.session.timeo.replacement_timeout = {{ open_iscsi_replacement_timeout }}
+node.conn[0].timeo.login_timeout = {{ open_iscsi_login_timeout }}
```
Comment (Mr-Pi) [inline templates/iscsid.conf.j2:42]: Please open a ticket, to introduce support for more than one node connection.

**Review by Mr-Pi — CHANGES_REQUESTED** [review]

File: templates/iscsid.conf.j2 (hunk 6 above)
Comment (SlothOfAnarchy) [reply templates/iscsid.conf.j2:42]: Done: https://github.com/stuvusIT/open-iscsi/issues/2

**Comment by SlothOfAnarchy** [comment]
Please rereview

**Review by Mr-Pi — APPROVED** [review]

**Pull request merged**